tests/user/login.test.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-undef */
+import request from 'supertest';
+import app from '../../src/app';
+import { connectDB } from '../../src/app';
+import {
+	verifiedLogin,
+	unverifiedLogin,
+	unregisteredLogin,
+	invalidPassword,
+} from '../mocks/user.mock';
+
+beforeAll(async () => {
+	await connectDB();
+
+});
+
+describe('POST /login', () => {
+	describe('Invalid credentials provided', () => {
+		describe('Invalid email', () => {
+			test('404 status and error message', async () => {
+				const response = await request(app)
+					.post('/api/v1/users/login')
+					.send(unregisteredLogin);
+				expect(response.body.error).toEqual('Wrong credentials');
+				expect(response.statusCode).toEqual(400);
+			});
+			describe('Unverified email', () => {
+				test('400 status and error message', async () => {
+					const response = await request(app)
+						.post('/api/v1/users/login')
+						.send(unverifiedLogin);
+					expect(response.body.error).toBeDefined();
+					expect(response.statusCode).toEqual(400);
+				});
+			});
+		});
+		describe('Invalid Password', () => {
+			test('400 status and wrong credentials', async () => {
+				const response = await request(app)
+					.post('/api/v1/users/login')
+					.send(invalidPassword);
+				expect(response.body.error).toEqual('Wrong credentials');
+				expect(response.statusCode).toEqual(400);
+			});
+		});
+	});
+	describe('Valid credentials provided', () => {
+		test('token, success message', async () => {
+			const response = await request(app)
+				.post('/api/v1/users/login')
+				.send(verifiedLogin);
+			expect(response.body.message).toEqual('Login successful');
```

Comment:
Can you reduce `describe` in this test? Like putting all tests in a single describe.